internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/integration/utils/VeniceTwoLayerMultiRegionMultiClusterWrapper.java

```diff
@@ -136,13 +136,16 @@ static ServiceProvider<VeniceTwoLayerMultiRegionMultiClusterWrapper> generateSer
       parentControllerProps.setProperty(PARTICIPANT_MESSAGE_STORE_ENABLED, "true");
       parentControllerPropertiesOverride = Optional.of(new VeniceProperties(parentControllerProps));
 
-      Map<String, String> clusterToD2 = new HashMap<>();
+      Map<String, String> clusterToRouterD2 = new HashMap<>();
+      Map<String, String> clusterToServerD2 = new HashMap<>();
       String[] clusterNames = new String[numberOfClustersInEachRegion];
       for (int i = 0; i < numberOfClustersInEachRegion; i++) {
         String clusterName = "venice-cluster" + i;
         clusterNames[i] = clusterName;
-        String d2ServiceName = "venice-" + i;
-        clusterToD2.put(clusterName, d2ServiceName);
+        String routerD2ServiceName = "venice-" + i;
+        clusterToRouterD2.put(clusterName, routerD2ServiceName);
+        String serverD2ServiceName = Utils.getUniqueString(clusterName + "_d2");
```

Comment:
Any reason not to use a deterministic pattern like the router's d2 service name? This will come back later when you try to write end to end test with the fast-client since the fast-client will need to set this service name systematically based on the cluster discovery result.

Reply:
We intentionally do not want to use deterministic service names. It leads to us hardcoding these names in main code later down the line

Reply:
Sure, but if we want to move away from hardcoded service names for d2 in client then we also need to offer a way to discover them. Whether the discovery is done by the controller or router we still need to provide it as a hardcoded string in a config or something. I guess the benefit of that is we have more control over it since it's abstracted away from the users/clients.

Reply:
Or do you mean also providing them in the form for client configs to the users?

Reply:
In a proper deployment, we control it via configs. For tests, we have `VeniceClusterWrapper#getServerD2ServiceName`